src/main/java/com/shade/decima/ui/data/viewer/texture/exporter/TextureExporterPNG.java

```diff
@@ -15,7 +17,13 @@ public class TextureExporterPNG implements TextureExporter {
     @Override
     public void export(@NotNull ImageProvider provider, @NotNull Set<Option> options, @NotNull WritableByteChannel channel) throws IOException {
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        ImageIO.write(provider.getImage(0, 0), "png", baos);
+        final BufferedImage concatImage = new BufferedImage(provider.getMaxWidth(), provider.getMaxHeight()*provider.getSliceCount(0), BufferedImage.TYPE_INT_ARGB);
+        final Graphics2D g2d = concatImage.createGraphics();
+        for (int slice = 0; slice < provider.getSliceCount(0); slice++) {
+            g2d.drawImage(provider.getImage(0, slice), 0, provider.getMaxHeight()*slice, null);
+        }
```

Comment:
Have you tested this for 3D textures? They could be handled too.

Reply:
Please take a look at the `system/textures/clouds/cloud3dnoisea/cloud3dnoisea.core` (use Edit -> Find Files to find it)